datafusion/substrait/src/logical_plan/producer.rs

```diff
@@ -109,6 +110,17 @@ pub fn to_substrait_rel(
             });
 
             if let Some(struct_items) = projection {
+                let names = match &scan.table_name {
+                    TableReference::Bare { table } => vec![table.to_string()],
+                    TableReference::Partial { schema, table } => {
+                        vec![schema.to_string(), table.to_string()]
+                    }
+                    TableReference::Full {
+                        catalog,
+                        schema,
+                        table,
+                    } => vec![catalog.to_string(), schema.to_string(), table.to_string()],
+                };
```

Comment:
could move this logic into a function on TableReference itself, since would future proof in case ever want to support nested namespaces in future

Reply:
:+1: I move it to `TableReference::to_vec()`